From oliver@neukum.org  Fri Feb  6 15:52:56 2009
From: Oliver Neukum <oliver@neukum.org>
Date: Fri, 6 Feb 2009 18:06:43 +0100
Subject: USB: serial: use generic method if no alternative is provided in usb serial layer
To: Greg KH <greg@kroah.com>
Message-ID: <200902061806.44184.oliver@neukum.org>
Content-Disposition: inline


This patch makes use of the generic method if a serial driver provides
no implementation. This simplifies implementing suspend/resume support
in serial drivers.

Signed-off-by: Oliver Neukum <oneukum@suse.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/usb-serial.c |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

--- a/drivers/usb/serial/usb-serial.c
+++ b/drivers/usb/serial/usb-serial.c
@@ -1085,12 +1085,15 @@ EXPORT_SYMBOL(usb_serial_suspend);
 int usb_serial_resume(struct usb_interface *intf)
 {
 	struct usb_serial *serial = usb_get_intfdata(intf);
+	int rv;
 
 	serial->suspending = 0;
 	if (serial->type->resume)
-		return serial->type->resume(serial);
+		rv = serial->type->resume(serial);
+	else
+		rv = usb_serial_generic_resume(serial);
 
-	return 0;
+	return rv;
 }
 EXPORT_SYMBOL(usb_serial_resume);
 
